R/update-news.R

```diff
@@ -19,39 +20,78 @@ collect_news <- function(messages) {
     purrr::discard(~ . == "") %>%
     purrr::map_chr(remove_housekeeping) %>%
     purrr::map(extract_newsworthy_items) %>%
-    unlist()
+    purrr::keep(~ nrow(.) > 0) %>%
+    bind_rows()
 
-  if (length(newsworthy_items) == 0) {
+  if (is.null(newsworthy_items)) {
     if (length(messages) <= 1) {
-      newsworthy_items <- "- Same as previous version."
+      newsworthy_items <- parse_regular_commit("Same as previous version.")
       if (fledge_chatty()) cli_alert_info("Same as previous version.")
     } else {
-      newsworthy_items <- "- Internal changes only."
+      newsworthy_items <- parse_regular_commit("Internal changes only.")
       if (fledge_chatty()) cli_alert_info("Internal changes only.")
     }
   } else {
     if (fledge_chatty()) {
-      no <- length(newsworthy_items)
+      no <- nrow(newsworthy_items)
       entry_word <- if (no == 1) "entry" else "entries"
       cli_alert_success(sprintf("Found %s NEWS-worthy %s.", no, entry_word))
     }
   }
 
-  paste0(paste(newsworthy_items, collapse = "\n"), "\n\n")
+  newsworthy_items
 }
 
 remove_housekeeping <- function(message) {
   strsplit(message, "\n---", fixed = TRUE)[[1]][1]
 }
 
 extract_newsworthy_items <- function(message) {
+  # Merge messages
+  if (is_merge_commit(message)) {
+    pr_data <- harvest_pr_data(message)
+    pr_number <- pr_data$pr_number
+    title <- if (is.na(pr_data$title)) {
+      sprintf("PLACEHOLDER https://github.com/%s/pull/%s", github_slug(), pr_number)
```

Comment:
I added a link because this way one could click on it.
Should I already add a function for replacing these items i.e. retrying to harvest PR data?

Reply:
The clickable link is a good idea. Let's postpone extensions for now.